plugins/inputs/snmp/snmp.go

```diff
@@ -0,0 +1,425 @@
+package snmp
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdb/telegraf/plugins/inputs"
+
+	"github.com/soniah/gosnmp"
+)
+
+// Snmp is a snmp plugin
+type Snmp struct {
+	Host              []Host
+	Get               []Data
+	Bulk              []Data
+	Snmptranslatefile string
+}
+
+type Host struct {
+	Address   string
+	Community string
+	// SNMP version. Default 2
+	Version int
+	// SNMP timeout, in seconds. 0 means no timeout
+	Timeout float64
+	// SNMP retries
+	Retries int
+	// Data to collect (list of Data names)
+	Collect []string
+	// Oids
+	getOids  []Data
+	bulkOids []Data
+}
+
+type Data struct {
+	Name string
+	// OIds
+	Oid string
+	// Unit
+	Unit string
+	//  SNMP getbulk max repetition
+	MaxRepetition uint8 `toml:"max_repetition"`
+	// We want to resolve oid to names oids text file
+	Snmptranslate bool
```

Comment:
Good question ! In fact this option define if you want to use the name from `Name` option in the section or if you want to use the name from `snmptranslate_file`.
So, maybe we can delete this option and do:
1. Get name from `snmptranslate_file` if it exists
2. Get name from `Name` option in the section
In that case we can not use OID as name (.1.3.6.....) which is good I think

Are you agree with that ?